src/window.rs

```diff
@@ -56,6 +56,48 @@ struct WindowWrapper {
     fullscreen: bool,
 }
 
+pub fn window_geometry() -> Result<(u64, u64), String> {
+    let prefix = "--geometry=";
+    let arg = std::env::args()
+        .filter(|arg| arg.starts_with(prefix))
+        .next();
+    match arg {
+        Some(sizes_str) => {
+            let input = sizes_str.split_at(prefix.len()).1;
+            let mut sizes = input.split('x');
+
+            let err_msg = || {
+                format!(
+                    "Invalid geometry: {}\nValid format: --geometry=<width>x<height>",
+                    input
+                )
+            };
+
+            let (width, height);
+            let mut parse_next = || {
+                sizes
+                    .next()
+                    .and_then(|s| s.parse::<u64>().ok())
+                    .ok_or_else(err_msg)
+            };
+            width = parse_next()?;
+            height = parse_next()?;
+
+            if sizes.next().is_some() {
+                Err(err_msg())
+            } else if width == 0 || height == 0 {
+                Err("Invalid geometry: Window dimensions should be greater than 0.".into())
```

Comment:
This error information doesn't appear to be surfaced to the user anywhere

Reply:
My preference is to surface it by an error! log which I believe is configured to output to stderr by default

Reply:
The error is returned to `main` in case of failure, then printed with `eprintln!`. Would you rather we exit the process from inside the function?

Reply:
Ah gotcha. No thats fine and I think your handling is reasonable.